ratelimitj-redis/src/test/java/es/moki/ratelimitj/redis/request/RedisSlidingWindowSyncRequestRateLimiterTest.java

```diff
@@ -1,28 +1,76 @@
 package es.moki.ratelimitj.redis.request;
 
+import com.google.common.collect.ImmutableSet;
 import es.moki.ratelimitj.core.limiter.request.RequestLimitRule;
 import es.moki.ratelimitj.core.limiter.request.RequestRateLimiter;
 import es.moki.ratelimitj.core.time.TimeSupplier;
 import es.moki.ratelimitj.redis.extensions.RedisStandaloneConnectionSetupExtension;
 import es.moki.ratelimitj.test.limiter.request.AbstractSyncRequestRateLimiterTest;
-import io.lettuce.core.RedisClient;
-import io.lettuce.core.api.StatefulRedisConnection;
-import org.junit.jupiter.api.AfterAll;
-import org.junit.jupiter.api.AfterEach;
-import org.junit.jupiter.api.BeforeAll;
+import es.moki.ratelimitj.test.time.TimeBanditSupplier;
+import java.time.Duration;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.stream.IntStream;
+import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.RegisterExtension;
 
 import java.util.Set;
-
+import static org.assertj.core.api.Assertions.assertThat;
 
 
 public class RedisSlidingWindowSyncRequestRateLimiterTest extends AbstractSyncRequestRateLimiterTest {
 
+    private final TimeBanditSupplier timeBandit = new TimeBanditSupplier();
+
     @RegisterExtension
     static RedisStandaloneConnectionSetupExtension extension = new RedisStandaloneConnectionSetupExtension();
 
     @Override
     protected RequestRateLimiter getRateLimiter(Set<RequestLimitRule> rules, TimeSupplier timeSupplier) {
         return new RedisSlidingWindowRequestRateLimiter(extension.getScriptingReactiveCommands(), extension.getKeyReactiveCommands(), rules, timeSupplier);
     }
+
```

Comment:
Added the integration tests here for now as the logic was implemented only for Redis.
After rolling it out to other modules we can move it to the super class.